src/bitcoin-mine.cpp

```diff
@@ -0,0 +1,124 @@
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <config/bitcoin-config.h> // IWYU pragma: keep
+
+#include <chainparams.h>
+#include <chainparamsbase.h>
+#include <clientversion.h>
+#include <common/args.h>
+#include <common/system.h>
+#include <compat/compat.h>
+#include <init/common.h>
+#include <interfaces/init.h>
+#include <interfaces/ipc.h>
+#include <logging.h>
+#include <tinyformat.h>
+#include <util/translation.h>
+
+static const char* const HELP_USAGE{R"(
+bitcoin-mine is a test program for interacting with bitcoin-node via IPC.
+
+Usage:
+  bitcoin-mine [options]
+)"};
+
+static const char* HELP_EXAMPLES{R"(
+Examples:
+  # Start separate bitcoin-node that bitcoin-mine can connect to.
+  bitcoin-node -regtest -ipcbind=unix
+
+  # Connect to bitcoin-node and print tip block hash.
+  bitcoin-mine -regtest
+
+  # Run with debug output.
+  bitcoin-mine -regtest -debug
+)"};
+
+const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+static void AddArgs(ArgsManager& args)
+{
+    SetupHelpOptions(args);
+    SetupChainParamsBaseOptions(args);
+    args.AddArg("-version", "Print version and exit", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    args.AddArg("-datadir=<dir>", "Specify data directory", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    args.AddArg("-ipcconnect=<address>", "Connect to bitcoin-node process in the background to perform online operations. Valid <address> values are 'unix' to connect to the default socket, 'unix:<socket path>' to connect to a socket at a nonstandard path. Default value: unix", ArgsManager::ALLOW_ANY, OptionsCategory::IPC);
+    init::AddLoggingArgs(args);
+}
+
+MAIN_FUNCTION
+{
+    ArgsManager& args = gArgs;
+    AddArgs(args);
+    std::string error_message;
+    if (!args.ParseParameters(argc, argv, error_message)) {
+        tfm::format(std::cerr, "Error parsing command line arguments: %s\n", error_message);
+        return EXIT_FAILURE;
+    }
+    if (!args.ReadConfigFiles(error_message, true)) {
+        tfm::format(std::cerr, "Error reading config files: %s\n", error_message);
+        return EXIT_FAILURE;
+    }
+    if (HelpRequested(args) || args.IsArgSet("-version")) {
+        std::string output{strprintf("%s bitcoin-mine version", PACKAGE_NAME) + " " + FormatFullVersion() + "\n"};
+        if (args.IsArgSet("-version")) {
+            output += FormatParagraph(LicenseInfo());
+        } else {
+            output += HELP_USAGE;
+            output += args.GetHelpMessage();
+            output += HELP_EXAMPLES;
+        }
+        tfm::format(std::cout, "%s", output);
+        return EXIT_SUCCESS;
+    }
+    if (!CheckDataDirOption(args)) {
+        tfm::format(std::cerr, "Error: Specified data directory \"%s\" does not exist.\n", args.GetArg("-datadir", ""));
+        return EXIT_FAILURE;
+    }
+    SelectParams(args.GetChainType());
```

Comment:
Can you move this above `AddArgs`? In https://github.com/Sjors/bitcoin/pull/48 I'm adding some argument documentation that need accsess to `BaseParams`.

Reply:
Never mind, this causes more headaches that it solves.